workflows/phaseimpute/main.nf

```diff
@@ -97,31 +99,44 @@ workflow PHASEIMPUTE {
     if (params.steps.split(',').contains("simulate") || params.steps.split(',').contains("all")) {
         // Output channel of simulate process
         ch_sim_output = Channel.empty()
+        // Set truth channel
+        ch_input_validate_truth = ch_input_sim
 
         // Test if the input are all bam files
         getAllFilesExtension(ch_input_sim)
             .map{ if (it != "bam") {
                 error "All input files must be in BAM format to perform simulation"
             } }
 
-        // Split the bam into the region specified
-        BAM_REGION(ch_input_sim, ch_region, ch_fasta)
-        ch_versions = ch_versions.mix(BAM_REGION.out.versions)
+        // Compute coverage of input files
+        SAMTOOLS_COVERAGE_TRT(ch_input_sim, ch_fasta)
+        ch_versions      = ch_versions.mix(SAMTOOLS_COVERAGE_TRT.out.versions)
+        ch_multiqc_files = ch_multiqc_files.mix(SAMTOOLS_COVERAGE_TRT.out.coverage.map{it[1]})
 
-        // Initialize channel to impute
-        ch_bam_to_impute = Channel.empty()
+        if (params.sim_by_reg == true) {
```

Comment:
Instead of adding a new params, is it possible to evaluate if the user thas set `--input_region != null`? I think our current behavior is that if the user does not add anything, then we process by full bam. I think in this case this is similar, and we can simplify by having less params.

Reply:
That could effectively be a solution.
I'll add that.